metals/src/main/scala/scala/meta/internal/metals/codeactions/UseNamedArguments.scala

```diff
@@ -0,0 +1,164 @@
+      if (symbol.endsWith(".") && !symbol.matches(methodSymbolPattern)) {
+        /*
+         * We've probably been given a companion object symbol
+         * e.g. in the case (@@ = cursor)
+         *
+         * case class Foo(a: Int, b: Int)
+         * val x = Fo@@o(1, 2)
+         *
+         * the definition provider gives us the symbol of the companion
+         * object. This is no good to us, so we try to replace it with
+         * the case class's symbol.
+         * In other words we turn "example/Foo." into "example/Foo#"
+         */
+        symbol.stripSuffix(".") ++ "#"
+      } else {
+        symbol
+      }
+    }
+
+    def buildEdits(
+        tree: Tree,
+        paramNames: List[String]
+    ): List[l.TextEdit] = {
+      val args = tree match {
+        case Term.Apply(_, xs) => xs
+        case Init(_, _, xss) => xss.flatten
+        case _ => Nil
+      }
+      args.zip(paramNames).flatMap {
+        case (Term.Assign(_, _), _) =>
+          // already a named argument, no edit needed
+          None
+        case (term, paramName) =>
+          val position = term.pos.toLSP.getStart
+          val text = s"$paramName = "
+          val edit = new l.TextEdit(new l.Range(position, position), text)
+          Some(edit)
+      }
+    }
+
+    val path = params.getTextDocument().getUri().toAbsolutePath
+
+    Future {
+      (for {
+        rootTree <- trees.get(path)
+        methodApplyTree <- findMethodApplyOrCtorTreeUnderCursor(
+          rootTree,
+          params.getRange
+        )
+        textDocument <- semanticdbs.textDocument(path).documentIncludingStale
+        symbolTree <- findSymbolTree(methodApplyTree)
+        resolvedSymbol = resolveSymbol(
+          params.getTextDocument,
+          path,
+          textDocument,
+          symbolTree.pos
+        )
+        symbolOccurrence <- resolvedSymbol.occurrence
+        symbol = tweakSymbol(symbolOccurrence.symbol)
+        symbolDocumentation <- symbolSearch
+          .documentation(symbol)
+          .asScala
+        parameterNames = symbolDocumentation
+          .parameters()
+          .asScala
+          .map(_.displayName())
+          .toList
+      } yield {
+        val codeEdits = buildEdits(
+          methodApplyTree,
+          parameterNames
+        )
+
+        codeEdits match {
+          case Nil => None // refactoring results in no changes to the code
+          case edits =>
+            val edit = new l.WorkspaceEdit()
+            val uri = params.getTextDocument().getUri()
+            val changes = Map(uri -> edits.asJava)
+
+            val codeAction = new l.CodeAction()
+            codeAction.setTitle(UseNamedArguments.title)
+            codeAction.setKind(l.CodeActionKind.Refactor)
+
+            edit.setChanges(changes.asJava)
+            codeAction.setEdit(edit)
+            Some(codeAction)
+        }
+      }).flatten.toSeq
+    }
+
+  }
+
+}
+
+object UseNamedArguments {
+
+  val title = "Use named arguments"
```

Comment:
```suggestion
  val label = "Use named arguments"
```

Reply:
I don't really mind either way, but FYI lsp4j (and the LSP spec) calls it a title.

Reply:
ah, nevermind then, can we rename the `ImpostMissingSymbol` one for consistency?